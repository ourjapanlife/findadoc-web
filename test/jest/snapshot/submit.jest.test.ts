import { jest } from '@jest/globals'
import { mount } from '@vue/test-utils'
import { createI18n } from 'vue-i18n'
import { createTestingPinia } from '@pinia/testing'
import Submit from '~/pages/submit.vue'

const mockSubmissionStore = {
    submissionCompleted: false,
    createNewSubmission: jest.fn(),
}

jest.mock('~/stores/submissionStore', () => ({
    useSubmissionStore: () => mockSubmissionStore,
}))

const NuxtLinkStub = {
    template: '<a><slot /></a>',
    props: ['to'],
}

const i18n = createI18n({
    legacy: false,
    locale: 'en-US',
    messages: { 'en-US': {} }
})

describe('submit.vue', () => {
    const pinia = createTestingPinia({
        createSpy: jest.fn,
    })

    it('renders correctly', () => {
        const wrapper = mount(Submit, {
            global: {
                plugins: [i18n, pinia],
                stubs: {
                    SubmissionCompleted: true,
                    Loader: true,
                    NuxtLink: NuxtLinkStub,
                },
            }
        })
        expect(wrapper.element).toMatchSnapshot()
    })
})
