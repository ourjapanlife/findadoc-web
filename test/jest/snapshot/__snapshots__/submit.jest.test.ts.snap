// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submit.vue renders correctly 1`] = `
<div
  class="flex flex-col h-full w-full place-self-center overflow-y-auto pb-8"
>
  <submissionform />
  <!--v-if-->
</div>
`;
